src/stan/callbacks/unique_stream_writer.hpp

```diff
@@ -114,6 +114,15 @@ class unique_stream_writer final : public writer {
     *output_ << comment_prefix_ << message << std::endl;
   }
 
+  /**
+   * Checks if stream is valid.
+   */
+  bool is_nonnull() const noexcept { return output_ != nullptr; }
```

Comment:
[optional] Almost all the overloads in these files start with an if, that could be re-written to use this helper